numpy/lib/arraysetops.py

```diff
@@ -627,22 +627,29 @@ def in1d(ar1, ar2, assume_unique=False, invert=False, *, kind=None):
     if ar2.dtype == bool:
         ar2 = ar2 + np.uint8(0)
 
-    # Check if we can use a fast integer algorithm:
-    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
-                      np.issubdtype(ar2.dtype, np.integer))
-
     if kind not in {None, 'sort', 'table'}:
         raise ValueError(
             f"Invalid kind: '{kind}'. Please use None, 'sort' or 'table'.")
 
-    if integer_arrays and kind in {None, 'table'}:
+    # Check if we can use a fast integer algorithm:
+    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
+                      np.issubdtype(ar2.dtype, np.integer))
```

Comment:
I am considering to use `ar1.dtype.kind in "ui?"` instead.  Which already rejects timedelta.  Can be (incorrectly) faked by downstream, but I think it is actually less likely to fail then `issubdtype`.

At some point `issubdtype` has probably a bit clearner logic and this would be the better way though, but then @eendebakpt won't be tempted to micro-optimize it away ;).

Reply:
(using an incorrect `i` or `u` kind lead to ugly problems until recent NumPy versions, so I don't expect anyone doing it seriously in the real world.)

Reply:
Sounds good to me. Should I add the key for boolean, since we're moving the bool->int conversion inside table (after this part)?

Reply:
Changed to:
```python
    is_integer_arrays = all(ar.dtype.kind in "uib?" for ar in (ar1, ar2))
```
and all the tests pass.